### PR TITLE
Add keyboard support for text fields 

### PR DESCRIPTION
**Pull request by roscale**
The keyboard can now be used to insert text into text fields created by the compositor itself in Flutter.
Key repeat also works as intended, not only when entering text, but also on keyboard shortcuts which can be useful if the user wants to hold a key combination like alt+tab to quickly browse through a list of items.

This PR also fixes an issue where modifier keys would get stuck after unfocusing the window.
However, this fix was implemented using a modified version of Smithay, and I'm awaiting for the following PR to get merged before merging this one:

https://github.com/Smithay/smithay/pull/1300

### CODE REVIEW

**Comment by roscale** [comment]
Sorry for the large diff on `keyboard.rs`. I just moved its contents to `glfw_key_codes.rs` but apparently Git considered it as a big insertion and deletion.

**Comment by roscale** [comment]
https://github.com/Smithay/smithay/pull/1300 has been merged, so it's all good. I've updated Cargo.toml to use the latest version from upstream.

File: embedder/src/main.rs
Comment (PapyElGringo) [inline embedder/src/main.rs:1]: I think our IDE are fighting for formating we should find a way to make them agree on it

**Review by PapyElGringo — APPROVED** [review]

File: embedder/src/main.rs
Comment (roscale) [reply embedder/src/main.rs:1]: Hmm, ok. Apparently my IDE defaults to its own formatting. I'll make sure to run rustfmt next time.

**Pull request merged**